File: P041/main.cpp
#include <iostream>
#include <chrono>
#include <ctime>
#include <math.h>

using namespace std;

bool is_prime(const int a) {
    if(a == 1) return false;

    for(long i=2; i<=sqrt(a); i++)
        if(a%i==0)
           return false;
    return true;
}

bool isPandigital(const int a) {
    string stringTemp = to_string(a);
    int temp[stringTemp.size()-1] = {0};

    for(unsigned i=0; i<stringTemp.size(); i++) {
        int yes = stringTemp[i]-49;
        if(temp[yes] == 1) {
            return false;
        }
        else {
            temp[yes] = 1;
        }
    }

    for(unsigned i=0; i<stringTemp.size()-1; i++) {
        if(temp[i] != 1) return false;
    }
    return true;
}

int main(int argc, const char**argv)  {
    typedef chrono::high_resolution_clock Clock;
    auto t1 = Clock::now();

    long temp = 0;

    for(long i=7654321; i>=2143; i-=2) {
        if(is_prime(i)) {
            if(isPandigital(i)) {
                cout << i << endl;
                break;
            }
        }
    }

    auto t2 = Clock::now();
    chrono::milliseconds ns = chrono::duration_cast<chrono::milliseconds>(t2-t1);
    cout << ns.count() << " milliseconds" << endl;
    return 0;
}

File: P041/main.java
import java.lang.*;
import java.math.*;

public class main {
    private static boolean is_prime(int a) {
        if(a == 1)
            return false;

        for(int i=2; i<=(int)Math.sqrt(a); i++)
            if(a%i==0)
                return false;
        return true;
    }

    private static boolean isPandigital(int a) {
        String stringTemp = String.valueOf(a);
        String[] list = new String[stringTemp.length()-1];

        for(int i=0; i<stringTemp.length(); i++) {
            int yes = stringTemp[i];
            System.out.println(yes);
        }

        return true;
    }

    public static void main(String[] args) {
        long startTime = System.currentTimeMillis();


        long endTime = System.currentTimeMillis();
        long totalSpan = System.currentTimeMillis();

        System.out.println("Completed in " + totalSpan + " milliseconds.");
    }
}
